### PR TITLE
Migrate farm text encoder 152

### LINKED ISSUE #152: migrate farm text encoder
┆Issue is synchronized with this [Jira Task](https://jinaai.atlassian.net/browse/JINAHUB-41) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by YueLiu-jina**
resolve #152

### CODE REVIEW

File: encoders/nlp/FarmTextEncoder/tests/test_farmtextencoder.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+
+# @pytest.mark.skipif('JINA_TEST_PRETRAINED' not in os.environ, reason='skip the pretrained test if not set')
+def test_encoding_results():
+    target_output_dim = 768
+    metas = get_metas()
+    encoder = FarmTextEncoder(metas=metas)
+    test_data = np.array(['it is a good day!', 'the dog sits on the floor.'])
+    encoded_data = encoder.encode(test_data)
+    print(encoder.__dict__)
```
Comment (nan-wang) [inline encoders/nlp/FarmTextEncoder/tests/test_farmtextencoder.py:34]:
```suggestion
```

File: encoders/nlp/FarmTextEncoder/tests/test_farmtextencoder.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    for file in tmp_files:
+        if file and os.path.exists(file):
+            if os.path.isfile(file):
+                os.remove(file)
+            elif os.path.isdir(file):
+                shutil.rmtree(file, ignore_errors=False, onerror=None)
+
+
+
+# @pytest.mark.skipif('JINA_TEST_PRETRAINED' not in os.environ, reason='skip the pretrained test if not set')
```
Comment (nan-wang) [inline encoders/nlp/FarmTextEncoder/tests/test_farmtextencoder.py:27]:
```suggestion
class MockInference():
    def load(*args, **kwargs):
        return self
        
    def extract_vectors(dicts, *args, **kwargs):
        return [{'vec': np.random.random(target_output_dim)}]
       

@mock.patch('farm.infer.Inferencer', return_value=MockInference())
```

File: encoders/nlp/FarmTextEncoder/tests/test_farmtextencoder.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    encoder.save_config()
+    assert os.path.exists(encoder.config_abspath)
+    encoder_loaded = BaseExecutor.load_config(encoder.config_abspath)
+    assert encoder_loaded.model_name == encoder.model_name
+    rm_files([encoder.config_abspath])
+
+
+def teardown_module():
+    mlflow_dir = 'mlruns'
+    rm_files([mlflow_dir])
```
Comment (nan-wang) [inline encoders/nlp/FarmTextEncoder/tests/test_farmtextencoder.py:65]: check the PR at https://github.com/jina-ai/jina-hub/pull/169/files

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: encoders/nlp/FarmTextEncoder/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+FROM jinaai/jina
```
Comment (JoanFM) [inline encoders/nlp/FarmTextEncoder/Dockerfile:1]: remember to use `:devel`

File: encoders/nlp/FarmTextEncoder/tests/test_farmtextencoder.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+def test_encoding_results():
+    target_output_dim = 768
+    metas = get_metas()
+    encoder = FarmTextEncoder(metas=metas)
+    test_data = np.array(['it is a good day!', 'the dog sits on the floor.'])
+    encoded_data = encoder.encode(test_data)
+    print(encoder.__dict__)
+    assert encoded_data.shape == (2, target_output_dim)
+
+# @pytest.mark.skipif('JINA_TEST_PRETRAINED' not in os.environ, reason='skip the pretrained test if not set')
```
Comment (JoanFM) [inline encoders/nlp/FarmTextEncoder/tests/test_farmtextencoder.py:37]: completely remove these lines

**Review by hanxiao — CHANGES_REQUESTED** [review]
need to add one unit test with real model not with mocked model, see our Slack discussion

**Comment by YueLiu-jina** [comment]
added non-local test for test_encoding_result

File: encoders/nlp/FarmTextEncoder/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+COPY . /workspace
+WORKDIR /workspace
+
+ENV JINA_TEST_PRETRAINED='true'
+
+# install the third-party requirements
+RUN pip install -r requirements.txt
+
+# for testing the image
+RUN pip install pytest pytest-mock mock && pytest
```
Comment (nan-wang) [inline encoders/nlp/FarmTextEncoder/Dockerfile:13]:
```suggestion
RUN pip install pytest pytest-mock mock && JINA_TEST_PRETRAINED='true' pytest
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**